lib/klaviyo/client.rb

```diff
@@ -1,74 +1,67 @@
-      kwargs = defaults.merge(kwargs)
-      
-      if kwargs[:email].to_s.empty? and kwargs[:id].to_s.empty?
-        raise KlaviyoError.new('You must identify a user by email or ID')
+    @@domain = 'https://a.klaviyo.com/api'
+    @@v1 = 'v1'
+    @@v2 = 'v2'
+
+    private
+
+    def self.request(method, path, kwargs = {})
+      if path == 'track' || path == 'identify'
+        params = build_params(kwargs)
+        url = "#{@@domain}/#{path}?#{params}"
+        res = Faraday.get(url)
+      else
+        check_private_api_key_exists()
+        url = "#{@@domain}/#{path}"
+        connection = Faraday.new(
+          url: url,
+          headers: {
+            'Content-type' => 'application/json'
+        })
+        response = connection.send(method.downcase) do |req|
+          req.body = kwargs[:body].to_json || nil
+        end
+        puts response.body
       end
-      
-      customer_properties = kwargs[:customer_properties]
-      customer_properties[:email] = kwargs[:email] unless kwargs[:email].to_s.empty?
-      customer_properties[:id] = kwargs[:id] unless kwargs[:id].to_s.empty?
+    end
 
-      params = {
-        :token => @api_key,
-        :event => event,
-        :properties => kwargs[:properties],
-        :customer_properties => customer_properties,
-        :ip => ''
-      }
-      params[:time] = kwargs[:time].to_time.to_i if kwargs[:time]
-     
-      params = build_params(params)
-      request('api/track', params)
+    def self.v1_request(method, path, kwargs = {})
+      defaults = {:page => nil, :count => nil, :since => nil, :sort => nil}
+      params = defaults.merge(kwargs)
+      query_params = encode_params(params)
+      url_params = "api_key=#{Klaviyo.private_api_key}#{query_params}"
+      full_path = "#{@@v1}/#{path}?#{url_params}"
+      request(method, full_path)
     end
-    
-    def track_once(event, opts = {})
-      opts.update('__track_once__' => true)
-      track(event, opts)
+
+    def self.v2_request(method, path, kwargs = {})
+      path = "#{@@v2}/#{path}"
+      key = {
+        "api_key": "#{@@private_api_key}"
+      }
+      data = {}
+      data[:body] = key.merge(kwargs)
+      request(method, path, data)
     end
-    
-    def identify(kwargs = {})
-      defaults = {:id => nil, :email => nil, :properties => {}}
-      kwargs = defaults.merge(kwargs)
-      
-      if kwargs[:email].to_s.empty? and kwargs[:id].to_s.empty?
-        raise KlaviyoError.new('You must identify a user by email or ID')
-      end
-      
-      properties = kwargs[:properties]
-      properties[:email] = kwargs[:email] unless kwargs[:email].to_s.empty?
-      properties[:id] = kwargs[:id] unless kwargs[:id].to_s.empty?
 
-      params = build_params({
-        :token => @api_key,
-        :properties => properties
-      })
-      request('api/identify', params)
+    def self.build_params(params)
+      "data=#{Base64.encode64(JSON.generate(params)).gsub(/\n/,'')}"
```

Comment:
You can get rid of the `gsub` call by using [`Base64.strict_encode64`](https://ruby-doc.org/stdlib-2.5.3/libdoc/base64/rdoc/Base64.html#method-i-strict_encode64)